saleor/product/views.py

```diff
@@ -113,8 +113,11 @@ def category_index(request, path, category_id):
     if actual_path != path:
         return redirect('product:category', permanent=True, path=actual_path,
                         category_id=category_id)
+    # Check for subcategories
+    categories = category.get_descendants(include_self=True)
+    ids = [c.id for c in categories]
```

Comment:
Can we use flat `values_list` on the queryset?

Reply:
Yes, we can!